docs/ibm-cloud-rules.md

```diff
@@ -2980,6 +2991,103 @@ paths:
 </table>
 
 
+### ibm-no-operation-requestbody
+<table>
+<tr>
+<td><b>Rule id:</b></td>
+<td><b>ibm-no-operation-requestbody</b></td>
+</tr>
+<tr>
+<td valign=top><b>Description:</b></td>
+<td>
+The <a href="https://cloud.ibm.com/docs/api-handbook?topic=api-handbook-methods#summary">API Handbook</a>
+provides guidance related to the use of various HTTP methods.
+In particular, the API Handbook discourages the use of a <code>requestBody</code>
+with <code>DELETE</code>, <code>GET</code>, <code>HEAD</code> and <code>OPTIONS</code> operations.
+This rule ensures that these operations do not define a <code>requestBody</code>.
+</td>
+</tr>
+<tr>
+<td><b>Severity:</b></td>
+<td>warn</td>
+</tr>
+<tr>
+<td><b>OAS Versions:</b></td>
+<td>oas3</td>
+</tr>
+<tr>
+<td valign=top><b>Configuration:</b></td>
+<td>This rule supports a configuration that specifies the set of HTTP methods that are checked
+for a <code>requestBody</code>.
+<p>The default configuration object provided with the rule is:
+<pre>
+{
+  httpMethods: ['delete', 'get', 'head', 'options']
+}
+</pre>
+<p>To configure the rule with a different set of HTTP methods, you'll need to
+<a href="#replace-a-rule-from-ibm-cloudopenapi-ruleset">replace this rule with a new rule within your
+custom ruleset</a> and modify the configuration such that the value of the <code>httpMethods</code> field 
+specifies the desired set of HTTP methods to be checked.
+For example, to enforce the rule for DELETE, HEAD and OPTIONS operations, the configuration object would look like this:
+<pre>
+{
+  httpMethods: ['delete', 'head', 'options']
+}
+</pre>
+</td>
+</tr>
+<tr>
+<td valign=top><b>Non-compliant example:<b></td>
+<td>
+<pre>
+paths:
+  '/v1/things/search':
+    get:      &lt;&lt;&lt; GET with requestBody
+      operationId: search_things
+      requestBody:
+        description: 'An object containing the search request.'
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/ThingSearchRequest'
+      responses:
+        '200':
+          description: 'Search completed successfully'
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ThingSearchResponse'
+</pre>
+</td>
+</tr>
+<tr>
+<td valign=top><b>Compliant example:</b></td>
+<td>
+<pre>
+paths:
+  '/v1/things/search':
+    post:     &lt;&lt;&lt; GET changed to POST
```

Comment:
Perhaps this should just be a GET without a request body? Not a big deal, I get what you are demonstrating here, just seemed odd to me that the operation switched

Reply:
haha, yeah I guess I was looking at it from the opposite angle... i.e. using a GET where we should really use a POST instead :)
I'll change it